test/e2e/e2e.go

```diff
@@ -2272,83 +2273,96 @@ func getNodePodCIDR(nodeName string) (string, error) {
 	const (
-		svcname                   string        = "delete-db"
-		ovnNs                     string        = "ovn-kubernetes"
-		databasePodPrefix         string        = "ovnkube-db"
-		databasePodNorthContainer string        = "nb-ovsdb"
-		northDBFileName           string        = "ovnnb_db.db"
-		southDBFileName           string        = "ovnsb_db.db"
-		dirDB                     string        = "/etc/ovn/"
-		waitingPeriod             time.Duration = 180 * time.Second // polling timeout
-		ovnWorkerNode             string        = "ovn-worker"
-		ovnWorkerNode2            string        = "ovn-worker2"
-		haModeMinDb               int           = 0
-		haModeMaxDb               int           = 2
+		svcname                   string = "delete-db"
+		ovnNs                     string = "ovn-kubernetes"
+		databasePodPrefix         string = "ovnkube-db"
+		databasePodNorthContainer string = "nb-ovsdb"
+		northDBFileName           string = "ovnnb_db.db"
+		southDBFileName           string = "ovnsb_db.db"
+		dirDB                     string = "/etc/ovn"
+		ovnWorkerNode             string = "ovn-worker"
+		ovnWorkerNode2            string = "ovn-worker2"
+		haModeMinDb               int    = 0
+		haModeMaxDb               int    = 2
 	)
 	var (
-		containerToDeleteFrom = databasePodNorthContainer
+		allDBFiles = []string{path.Join(dirDB, northDBFileName), path.Join(dirDB, southDBFileName)}
 	)
 	f := framework.NewDefaultFramework(svcname)
 
+	// WaitForPodConditionAllowNotFoundError is a wrapper for WaitForPodCondition that allows at most 6 times for the pod not to be found.
+	WaitForPodConditionAllowNotFoundErrors := func(f *framework.Framework, ns, podName, desc string, timeout time.Duration, condition podCondition) error {
+		max_tries := 6               // 6 tries to waiting for the pod to restart
+		cooldown := 10 * time.Second // 10 sec to cooldown between each try
+		for i := 0; i < max_tries; i++ {
+			err := e2epod.WaitForPodCondition(f.ClientSet, ns, podName, desc, 5*time.Minute, condition)
+			if apierrors.IsNotFound(err) {
+				// pod not found,try again after cooldown
+				time.Sleep(cooldown)
+				continue
+			}
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		return fmt.Errorf("Gave up after waiting %v for pod %q to be %q: pod is not found", timeout, podName, desc)
+	}
+
+	// waitForPodToFinishFullRestart waits for a the pod to finish it's reset cycle and returns.
+	waitForPodToFinishFullRestart := func(f *framework.Framework, pod *v1.Pod) {
+		err := e2epod.WaitForPodCondition(f.ClientSet, pod.Namespace, pod.Name, "not ready", 5*time.Minute, testutils.PodNotReady)
+		if err != nil {
+			framework.Failf("pod %v is not arrived to not-ready state: %v", pod.Name, err)
+		}
+		// during this stage on the restarting process we can encounter "pod not found" errors.
+		// this type of errors is valid because the pod is during restart so it will have a period that it will not be available.
```

Comment:
"these types of errors are valid because the pod is restarting so there will be a period of time it is unavailable"

Reply:
Thanks, Done